R/resp-stream.R

```diff
@@ -263,24 +297,27 @@ resp_boundary_pushback <- function(resp, max_size, boundary_func, include_traile
       # one extra byte so we know to error.
       n = min(chunk_size, max_size - length(buffer) + 1)
     )
-
     print_buffer(chunk, "Received chunk")
 
-    # If we've reached the end of input, store the buffer and return NULL
     if (length(chunk) == 0) {
       if (!isIncomplete(resp$body)) {
         # We've truly reached the end of the connection; no more data is coming
-        if (include_trailer && length(buffer) > 0) {
-          return(buffer)
-        } else {
+        if (length(buffer) == 0) {
```

Comment:
@jcheng5 I don't think I've changed the logic here, just tweaked it a bit so we can warn when we get an unexpected end of input.